File: src/3127.make-a-square-with-the-same-color.3398/solution.ts
function canMakeSquare (grid: string[][]): boolean {
    const N = grid.length;
    for (let i = 0; i < N - 1; i++) {
        for (let j = 0; j < N - 1; j++) {
            let cnt = 0;
            if (grid[i][j] === 'B') {
                cnt++;
            }
            if (grid[i + 1][j] === 'B') {
                cnt++;
            }
            if (grid[i][j + 1] === 'B') {
                cnt++;
            }
            if (grid[i + 1][j + 1] === 'B') {
                cnt++;
            }
            if (cnt !== 2) {
                return true;
            }
        }
    }
    return false;
}

File: src/3127.make-a-square-with-the-same-color.3398/solution.java
class Solution {
    public boolean canMakeSquare(char[][] grid) {
        for(int i=0;i<2;i++){
            for(int j=0;j<2;j++){
                int count = 0;
                if(grid[i][j] == 'B'){
                    count++;
                }
                if(grid[i+1][j] == 'B'){
                    count++;
                }
                if(grid[i][j+1] == 'B'){
                    count++;
                }
                if(grid[i+1][j+1] == 'B'){
                    count++;
                }
                if(count != 2){
                    return true;
                }
            }
        }
        return false;
    }
}